apps/modernization-ui/src/apps/page-builder/pages/EditPage/EditPageTabs/EditPageTabs.tsx

```diff
@@ -83,10 +78,28 @@ export const EditPageTabs = ({ tabs, active, setActive, onAddSuccess }: Props) =
         }
     };
 
+    const handleDeleteTab = async (tab: PagesTab) => {
+        setSelectedForDelete(undefined);
+        setSelectedEditTab(tab);
+        if (pageId) {
+            deleteTab(token, parseInt(pageId), tab.id!)
+                .then(() => {
+                    setDeleteSuccess(true);
+                    setIsEditing(false);
+                    onAddSuccess();
+                })
+                .then(() => setTimeout(() => resetEditPageTabs(), 3000));
+        }
+    };
+
     const resetEditPageTabs = () => {
         setSelectedEditTab(undefined);
+        setSelectedForDelete(undefined);
+        setIsAdding(false);
+        setIsEditing(false);
         setAddSuccess(false);
         setEditSuccess(false);
+        setDeleteSuccess(false);
```

Comment:
Suggestion: could potentially combine these into a singular state variable.

```typescript 
const [state, setState] = useState<'adding' | 'editing' | 'add-success' | 'edit-success' | 'delete-success'>('adding');
```

I haven't thought it out completely, but it seems we are tracking a lot of separate state variables. Could also be a good use case for `useReducer`

Reply:
I had tried `useReducer` here but the number of states needed to track the various editing/deleting/submitting phases started to balloon. Found `setState` to be more concise in this instance.